relay-protocol/src/annotated.rs

```diff
@@ -117,6 +117,26 @@ impl<T> Annotated<T> {
         Annotated(Some(value), Meta::default())
     }
 
+    /// Creates a new annotated value with meta data configured by the provided closure.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use relay_protocol::{Annotated, Meta};
+    /// let annotated = Annotated::new_with_meta(42, |meta| {
+    ///     meta.set_original_length(Some(10));
+    /// });
+    /// ```
+    #[inline]
+    pub fn new_with_meta<F>(value: T, f: F) -> Self
```

Comment:
I added this new helper to make the logic a bit easier, can revert this if folks feel like it's not needed.